feature_engine/imputation/base_imputer.py

```diff
@@ -87,17 +88,46 @@ def _get_feature_names_in(self, X):
 
         return self
 
-    def get_feature_names_out(self) -> List:
+    @Substitution(
+        input_features=_input_features_docstring,
+    )
+    def get_feature_names_out(
+            self, input_features: Optional[Union[List, str]] = None
+    ) -> List:
         """Get output feature names for transformation.
 
+        Parameters
+        ----------
+        {input_features}
+
         Returns
         -------
         feature_names_out: list
             The feature names.
         """
         check_is_fitted(self)
 
-        return self.feature_names_in_
+        if input_features is None:
+            # return all feature names
+            feature_names = [
+                f for f in self.feature_names_in_ if f not in self.features_to_drop_
```

Comment:
I dont think these classes have a fetures_to_drop attribute

Reply:
When/why do we create `self.features_to_drop_` for certain classes and no others?

Reply:
that attribute is only for the feature selection classes